src/System.Text.Utf8/src/System/Text/Utf8/Utf8String.cs

```diff
@@ -175,6 +188,11 @@ public override string ToString()
             return new string(characters);
         }
 
+        public bool ReferenceEquals(Utf8String other)
```

Comment:
I personally like it although I would call it "IsIdentical"/"IsSame" or something around that to not confuse it with reference type